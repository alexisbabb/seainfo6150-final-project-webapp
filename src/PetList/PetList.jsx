import React from "react";
import PropTypes from "prop-types";
import styles from "./PetList.module.css";
import PetListItem from "../PetListItem/PetListItem.jsx";

const PetList = (props) => {
  let displayContent;

  if (props.articles.length) {
    displayContent = (
      <ul className={styles.container}>
        {props.articles.map((article) => (
          <PetListItem article={article} key={article.slug} />
        ))}
      </ul>
    );
  } else {
    displayContent = <div>You have no data!</div>;
  }

  return (
    <div>
      {displayContent}
    </div>
  );
};

PetList.propTypes = {
    articles: PropTypes.array.isRequired
}
export default PetList;