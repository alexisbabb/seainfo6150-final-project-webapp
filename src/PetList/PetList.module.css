.container {
    background-image: linear-gradient(to bottom, rgba(255, 255, 0, 0.25), rgba(255, 0, 0, 0.25)); 
    display: grid;
    grid-template-columns: 1fr;
    gap: 20px;
    font-family: Arial, Helvetica, sans-serif;
    list-style-type: none;
    padding: 15px;
  }
  
  @media (min-width: 600px) {
    .container {
      grid-template-columns: repeat(2, 1fr);
    }
  }
  
  @media (min-width: 800px) {
    .container {
      grid-template-columns: repeat(3, 1fr);
    }}
  
  @media (min-width: 1200px) {
    .container {
      grid-template-columns: repeat(4, 1fr);
    }}