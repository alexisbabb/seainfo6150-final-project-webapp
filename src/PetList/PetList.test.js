import React from "react";
import PetList from "./PetList"; 

describe("PetLists tests", () => {
    it("renders correctly", () => {
        const petList = [
            {
                "slug": "00001",
                "name": "Aladdin",
                "species": "Cat", 
                "breed": "Domestic Short Hair", 
                "gender": "Male", 
                "age": "2 Years", 
                "size": "Medium", 
                "color": "Black", 
                "location": "Newark", 
                "bio": "<p>Meet Aladdin! This sweet boy is looking for the perfect lap to relax in! He will follow you everywhere you go just like a dog. He is a very friendly boy that gets along with others cats. He originally came from a large cat colony living outside in lower Delaware. Aladdin is FIV positive, but that doesn't mean he won't live a long, normal life.</p>",
                "image": {
                    "url": "https://pbs.twimg.com/media/DlmBBTXXcAEdn6V.jpg",
                    "height": "400",
                    "width": "400"
                } 
            },
            {
                "slug": "00002",
                "name": "Chloe",
                "species": "Dog", 
                "breed": "Labrador Retriever", 
                "gender": "Female", 
                "age": "1 Year", 
                "size": "Medium", 
                "color": "Black", 
                "location": "Pike Creek", 
                "bio": "<p>Meet Chloe! She is loving and fun, but needs a good amount cuddle time due to her past. She loves to run free through the yard and chase the birds. Chloe is also super respectful to everyone around her.</p>",
                "image": {
                    "url": "https://www.fananimal.com/wp-content/uploads/2019/03/fan-animal-los-perros-en-verdad-sienten-celos.jpg",
                    "height": "400",
                    "width": "400"
                }
            }
        ]; 
        const { container } = render(<PetList articles={petList}/>);
        expect(container).toMatchSnapshot();
    });
});