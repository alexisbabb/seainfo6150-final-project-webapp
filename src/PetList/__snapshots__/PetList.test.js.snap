// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetLists tests renders correctly 1`] = `
<div>
  <div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <article
          class="article"
        >
          <header
            class="header"
          >
            <img
              alt="Aladdin"
              class="image"
              src="https://pbs.twimg.com/media/DlmBBTXXcAEdn6V.jpg"
            />
            <h2>
              <a
                class="link"
                href="/petlist/00001"
              >
                Aladdin
              </a>
            </h2>
          </header>
          <div
            class="wrapper"
          />
          <button
            class="detailsButton"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="container"
      >
        <article
          class="article"
        >
          <header
            class="header"
          >
            <img
              alt="Chloe"
              class="image"
              src="https://www.fananimal.com/wp-content/uploads/2019/03/fan-animal-los-perros-en-verdad-sienten-celos.jpg"
            />
            <h2>
              <a
                class="link"
                href="/petlist/00002"
              >
                Chloe
              </a>
            </h2>
          </header>
          <div
            class="wrapper"
          />
          <button
            class="detailsButton"
          >
            Show more
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
